x-pack/plugin/data-frame/src/test/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformConfigTests.java

```diff
@@ -143,6 +152,30 @@ public void testPreventHeaderInjection() throws IOException {
                 () -> createDataFrameTransformConfigFromString(pivotTransform, "test_header_injection"));
     }
 
+    public void testSetIdInBody() throws IOException {
+        String pivotTransform = "{"
+                + " \"id\" : \"body_id\","
+                + " \"source\" : \"src\","
+                + " \"dest\" : \"dest\","
+                + " \"pivot\" : {"
+                + " \"group_by\": {"
+                + "   \"id\": {"
+                + "     \"terms\": {"
+                + "       \"field\": \"id\""
+                + "} } },"
+                + " \"aggs\": {"
+                + "   \"avg\": {"
+                + "     \"avg\": {"
+                + "       \"field\": \"points\""
+                + "} } } } }";
+
+        DataFrameTransformConfig dataFrameTransformConfig = createDataFrameTransformConfigFromString(pivotTransform, "body_id");
+        assertEquals("body_id", dataFrameTransformConfig.getId());
+
+        expectThrows(IllegalArgumentException.class,
+                () -> createDataFrameTransformConfigFromString(pivotTransform, "other_id"));
```

Comment:
You could also assert that the error message is as expected here.  (For example this will detect if the body ID and URL ID are the wrong way around when the error message is formed.)